vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c

```diff
@@ -1072,29 +1074,49 @@ WIFIReturnCode_t WIFI_Ping( uint8_t * pucIPAddr,
     return eWiFiNotSupported;
 }
 /*-----------------------------------------------------------*/
-
 WIFIReturnCode_t WIFI_GetIP( uint8_t * pucIPAddr )
 {
     WIFIReturnCode_t xRetVal = eWiFiFailure;
 
     if (pucIPAddr == NULL) {
         return xRetVal;
     }
-#if !AFR_ESP_LWIP
     /* Try to acquire the semaphore. */
     if( xSemaphoreTake( xWiFiSem, xSemaphoreWaitTicks ) == pdTRUE )
     {
+#if !AFR_ESP_LWIP
         *( ( uint32_t * ) pucIPAddr ) = FreeRTOS_GetIPAddress();
         xRetVal = eWiFiSuccess;
+#else /* running lwip */
+        tcpip_adapter_ip_info_t ipInfo;
+        int ret;
+
+        ret = tcpip_adapter_get_ip_info(TCPIP_ADAPTER_IF_STA, &ipInfo);
+        if (ret == ESP_OK)
+        {
+            xRetVal = eWiFiSuccess;
+            memcpy( pucIPAddr, &ipInfo.ip.addr, sizeof( ipInfo.ip.addr ) );
+            configPRINTF(("%s: local ip address is %d.%d.%d.%d\n",
```

Comment:
Why are we printing the IP address for lwip but not for FreeRTOS+TCP (after line 1089)?